tools/src/main/java/org/apache/kafka/tools/StreamsCleanupClient.java

```diff
@@ -0,0 +1,208 @@
+                zkUtils.close();
+            }
+            System.exit(exitCode);
+        }
+    }
+
+    private void parseArguments(final String[] args) throws IOException {
+        final OptionParser optionParser = new OptionParser();
+        applicationIdOption = optionParser.accepts("application-id", "The Kafka Streams application ID (application.id)")
+                .withRequiredArg()
+                .describedAs("id")
+                .ofType(String.class)
+                .required();
+        bootstrapServerOption = optionParser.accepts("bootstrap-server", "Format: <host:port>")
+                .withRequiredArg()
+                .describedAs("url")
+                .defaultsTo("localhost:9092")
+                .ofType(String.class);
+        zookeeperOption = optionParser.accepts("zookeeper", "Format: <host:port>")
+                .withRequiredArg()
+                .describedAs("url")
+                .defaultsTo("localhost:2181")
+                .ofType(String.class);
+        sourceTopicsOption = optionParser.accepts("source-topics", "Comma separated list of user source topics")
+                .withRequiredArg()
+                .describedAs("list")
+                .ofType(String.class);
+        intermediateTopicsOption = optionParser.accepts("intermediate-topics", "Comma separated list of intermediate user topics")
+                .withRequiredArg()
+                .describedAs("list")
+                .ofType(String.class);
+
+        try {
+            options = optionParser.parse(args);
+        } catch (OptionException e) {
+            optionParser.printHelpOn(System.err);
+            throw e;
+        }
+    }
+
+    private void resetSourceTopicOffsets() {
+        final List<String> topics = new LinkedList<>();
+        topics.addAll(options.valuesOf(sourceTopicsOption));
+        topics.addAll(options.valuesOf(intermediateTopicsOption));
+
+        if (topics.size() == 0) {
+            System.out.println("No source or intermediate topics specified.");
+            return;
+        }
+
+        System.out.println("Resetting offsets to zero for topics " + topics);
+
+        final Properties config = new Properties();
+        config.setProperty(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, options.valueOf(bootstrapServerOption));
+        config.setProperty(ConsumerConfig.GROUP_ID_CONFIG, options.valueOf(applicationIdOption));
+        config.setProperty(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false");
+
+        final KafkaConsumer<byte[], byte[]> client = new KafkaConsumer<>(config, new ByteArrayDeserializer(), new ByteArrayDeserializer());
+        try {
+            client.subscribe(topics);
+            client.poll(1);
+
+            for (TopicPartition partition : client.assignment()) {
+                client.seek(partition, 0);
+            }
+            client.commitSync();
+        } catch (RuntimeException e) {
+            System.err.println("Resetting source topic offsets failed");
```

Comment:
Also, in this case, it might be helpful to also log which topics we attempted to reset.  Otherwise the user may not know where to begin troubleshooting.